packages/mdc-list/foundation.js

```diff
@@ -249,6 +249,9 @@ class MDCListFoundation extends MDCFoundation {
       this.focusLastElement();
     } else if (isEnter || isSpace) {
       if (isRootListItem) {
+        // Return early if enter key is pressed on anchor element which triggers synthetic MouseEvent event.
+        if (evt.target.tagName === 'A' && isEnter) return;
```

Comment:
I suspect we shouldn't be doing this in the foundation for the sake of framework wrappers. The adapter API that's removed in this PR was also separating that from the foundation.